lib/datadog/appsec/remote.rb

```diff
@@ -74,12 +74,15 @@ def receivers(telemetry)
               case content.path.product
               when 'ASM_DD'
                 rules << parsed_content
+                content.applied
               when 'ASM_DATA'
                 data << parsed_content['rules_data'] if parsed_content['rules_data']
+                content.applied
               when 'ASM'
                 overrides << parsed_content['rules_override'] if parsed_content['rules_override']
                 exclusions << parsed_content['exclusions'] if parsed_content['exclusions']
                 custom_rules << parsed_content['custom_rules'] if parsed_content['custom_rules']
+                content.applied
```

Comment:
I think this place is not protecting us from have content marked as applied, but have it failed later this method.

I see that as in Kafka, you ACK if you're done with incoming changes/data and you really applied it. Hence we really apply the new configuration only on line 109 with a call to `reconfigure`, hence the success of that operation should signal us about successful content use.

Reply:
you are right, fixed that [d9bcea9](https://github.com/DataDog/dd-trace-rb/pull/4345/commits/d9bcea98ceca89e7382060d2988ea5a685fe28f1)